deltachat-ffi/deltachat.h

```diff
@@ -2722,7 +2747,11 @@ dc_context_t*  dc_accounts_get_account          (dc_accounts_t* accounts, uint32
  *     unmanaged account-context as created by dc_context_new().
  *     Once you do no longer need the context-object, you have to call dc_context_unref() on it,
  *     which, however, will not close the account but only decrease a reference counter.
+ *
  *     If there is no selected account, NULL is returned.
+ *
+ *     If the selected account is encrypted and not yet loaded using
```

Comment:
If I selected an encrypted account for which the passphrase is not stored in the keychain, then close DC and reopen it, selected account returns NULL and there is no way to tell which account is selected.

With the current API it returned some account which had an ID but was closed, so UI could ask the user for the passphrase to unlock the last selected account.

Since this API is not useful now, I think it should be removed too and replaced with `dc_accounts_get_selected_account_id` instead. Then UI can get the ID and try to get the account with such ID. If it is in the list of locked accounts, ask for the passphrase.

Reply:
good catch, shame this is yet a little more api churn but seems like the nicest solution.  i guess as a bonus it reduces the number of ways you can get a context object.